core/iwasm/interpreter/wasm_mini_loader.c

```diff
@@ -4518,25 +4520,30 @@ wasm_loader_prepare_bytecode(WASMModule *module, WASMFunction *func,
 
             case WASM_OP_RETURN:
             {
-                int32 idx;
-                uint8 ret_type;
-                for (idx = (int32)func->func_type->result_count - 1; idx >= 0; idx--) {
-                    ret_type = *(func->func_type->types
-                                 + func->func_type->param_count + idx);
-                    POP_TYPE(ret_type);
+                handle_op_return:
+                {
+                    int32 idx;
+                    uint8 ret_type;
+                    for (idx = (int32)func->func_type->result_count - 1; idx >= 0;
+                         idx--) {
+                        ret_type = *(func->func_type->types
+                                     + func->func_type->param_count + idx);
+                        POP_TYPE(ret_type);
 #if WASM_ENABLE_FAST_INTERP != 0
                     /* emit the offset after return opcode */
                     POP_OFFSET_TYPE(ret_type);
 #endif
```

Comment:
In fast-interpreter mode, when we jump into this branch from `WASM_OP_RETURN_CALL` the `POP_OFFSET_TYPE(ret_type)` will emit operand, which will make the generated code incorrect.

Maybe we need a flag to avoid `POP_OFFSET_TYPE(ret_type)` if we're checking result of tail call:
```suggestion
#if WASM_ENABLE_FAST_INTERP != 0
                    if (!tail_call_return_check)
                        /* emit the offset after return opcode */
                        POP_OFFSET_TYPE(ret_type);
#endif
```
and reset the flag after the switch block

Reply:
Postpone this part to fast-interpreter implementation.